src/api.js

```diff
@@ -49,13 +68,38 @@ const getCsp = async (siteName) => {
     return await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/netlify-toml`);
 }
 
+const createPageData = async ({folderName, subfolderName, newFileName, siteName, resourceName}, content) => {
+    const apiEndpoint = getPageApiEndpoint({folderName, subfolderName, newFileName, siteName, resourceName})
+    const params = { content }
+    await axios.post(apiEndpoint, params)
+    
+    // redirect to new page upon successful creation
+    if (folderName) {
+        return `/sites/${siteName}/folder/${folderName}/${encodeURIComponent(`${subfolderName ? `${subfolderName}/` : ''}${newFileName}`)}`
+    } 
+    if (resourceName) {
+        return `/sites/${siteName}/resources/${resourceName}/${encodeURIComponent(newFileName)}`
+    }
+    return `/sites/${siteName}/pages/${encodeURIComponent(newFileName)}`
+}
+
+const renamePageData = async ({folderName, subfolderName, fileName, siteName, resourceName, newFileName}, content, sha) => {
+    const apiEndpoint = getPageApiEndpoint({folderName, subfolderName, fileName, siteName, resourceName, newFileName})
```

Comment:
Can we simplify the logic here and reduce the responsibility of `getPageApiEndpoint`? For example, for `renamePageData`, we can simply use  `getRenamePageApiEndpoint` instead of calling `getPageApiEndpoint`, which then calls `getRenamePageApiEndpoint`. Likewise for `getCreatePageApiEndpoint`

Reply:
fixed 7911b8d